service-base/src/main/java/org/eclipse/hono/service/management/device/DeviceManagementHttpEndpoint.java

```diff
@@ -100,29 +113,43 @@ public void addRoutes(final Router router) {
                 .handler(this::doDeleteDevice);
     }
 
+    /**
+     * The service to forward requests to.
+     *
+     * @return The service to bind to, must never return {@code null}.
+     */
+    protected abstract DeviceManagementService getService();
+
     private void doGetDevice(final RoutingContext ctx) {
 
         final String deviceId = getDeviceIdParam(ctx);
         final String tenantId = getTenantParam(ctx);
         final HttpServerResponse response = ctx.response();
 
-        final JsonObject requestMsg = EventBusMessage.forOperation(RegistryManagementConstants.ACTION_GET)
-                .setTenant(tenantId)
-                .setDeviceId(deviceId)
-                .toJson();
+        final Span span = Util.newChildSpan(SPAN_NAME_GET_DEVICE, TracingHandler.serverSpanContext(ctx), tracer, tenantId, deviceId, getClass().getSimpleName());
```

Comment:
This span, and the other spans in this class, doesn't get finished. This is an issue in the `EventBus[*]ManagementAdapter` classes as well. I've created #1702 to fix that in the event bus classes (which we still might want to keep for Hono 1.1, see comment below). You could have a look there.

Reply:
Thanks for the suggestions. I tried to add it to this implementation. Can you take another look and make sure it's all there?

Reply:
I've had a look at `AbstractDeviceManagementHttpEndpoint` and the spans still need to get finished there. The places where to insert `span.finish()` would be the premature method exits, e.g. before the `return` here:
````
            TracingHelper.logError(span, "missing body");
            HttpUtils.badRequest(ctx, "missing body");
            return;
````
and also in the result handlers of the service invocations (e.g. after `writeOperationResponse`).

Reply:
Good catch. I think the param checks have to be done before span is created. At least that's consistent with the current implementation. I also cleaned it up a bit. Can you give it another look?